client/app/components/template-explorer/template-explorer.component.js

```diff
@@ -0,0 +1,199 @@
+/* eslint camelcase: "off" */
+export const TemplateExplorerComponent = {
+  controller: ComponentController,
+  controllerAs: 'vm',
+  templateUrl: 'app/components/template-explorer/template-explorer.html',
+};
+
+/** @ngInject */
+function ComponentController($filter, lodash, ListView, Language, TemplatesService, EventNotifications, Session, RBAC, ModalService,
+                             CollectionsApi, sprintf, Polling) {
+  const vm = this;
+  vm.$onInit = activate();
+  vm.$onDestroy = function() {
+    Polling.stop('templateListPolling');
+  };
+
+  function activate() {
+    angular.extend(vm, {
+      currentUser: Session.currentUser(),
+      loading: false,
+      templates: [],
+      limit: 20,
+      filterCount: 0,
+      filters: [],
+      templatesList: [],
+      selectedItemsList: [],
+      limitOptions: [5, 10, 20, 50, 100, 200, 500, 1000],
+      selectedItemsListCount: 0,
+      resolveTemplates: resolveTemplates,
+      updatePagination: updatePagination,
+      actionConfig: getActionConfig(),
+      menuActions: getMenuActions(),
+      toolbarConfig: getToolbarConfig(),
+      listConfig: getListConfig(),
+      sortConfig: getSortConfig(),
+      offset: 0,
+      pollingInterval: 10000,
+    });
+
+    resolveTemplates(vm.limit, 0);
+    Polling.start('templateListPolling', pollTemplates, vm.pollingInterval);
+  }
+
+  function getActionConfig() {
```

Comment:
I left this one in because its going to be used and filled out after this PR. It will get filled out in the next PR but this one had to get through first.